pkg/api/bind.go

```diff
@@ -104,54 +104,62 @@ func (s *server) bind(w http.ResponseWriter, r *http.Request) {
 
 	// Unpack the parameter map in the request to a struct
 	bindingParameters := serviceManager.GetEmptyBindingParameters()
-	decoderConfig := &mapstructure.DecoderConfig{
-		TagName: "json",
-		Result:  bindingParameters,
-	}
-	decoder, err := mapstructure.NewDecoder(decoderConfig)
-	if err != nil {
-		logFields["error"] = err
-		log.WithFields(logFields).Error(
-			"error building parameter map decoder",
-		)
-		s.writeResponse(w, http.StatusInternalServerError, generateEmptyResponse())
-		return
-	}
-	err = decoder.Decode(bindingRequest.Parameters)
-	if err != nil {
-		log.WithFields(logFields).Debug(
-			"bad binding request: error decoding parameter map",
-		)
-		// krancour: Choosing to interpret this scenario as a bad request since the
-		// probable cause would be disagreement between provided and expected types
-		s.writeResponse(w, http.StatusBadRequest, generateEmptyResponse())
-		return
+	if bindingParameters != nil {
```

Comment:
Only change to this whole block of code is the nil guard at the beginning. Why? Because I don't want to force managers that don't use any bindingParameters to have to provide a useless type for that purpose just because.